packages/mds-agency/request-handlers.ts

```diff
@@ -494,11 +495,14 @@ export const submitVehicleTelemetry = async (
       })
     }
   } catch (err) {
-    res.status(500).send({
-      error: 'server_error',
-      error_description: 'None of the provided data was valid',
-      error_details: [`device_id ${data[0].device_id}: not found`]
-    })
+    if (isError(err, NotFoundError))
+      return res.status(400).send({
+        error: 'unregistered',
+        error_description: 'Some of the devices are unregistered',
```

Comment:
Error message could be different since we know this is a singular device.

Reply:
Agreed, but this is 100% in alignment with the way that the Agency spec says to do things :|